### PR TITLE
feat: Safety alerts channel and mention raid protection

### PR DESCRIPTION
**Pull request by jaw0r3k**
**Please describe the changes this PR makes and why it should be merged:**
 Backports:
 -  #8959
 
 
### Summary
🛡️🗣️🚨

This pull request adds two new features to the discord.js library: mention raid protection and safety alerts channel. Mention raid protection allows the user to create and edit rules that automatically moderate spam attacks that mention many users or roles in a guild. Safety alerts channel allows the user to access and edit the channel where Discord sends notifications about potential threats to the guild's security. The pull request modifies the `AutoModerationRuleManager`, `AutoModerationRule`, and `Guild` classes, and the `GuildEditData` type, to implement these features.

> _`AutoModeration`_
> _Protects from mention raids_
> _A new feature blooms_

### Walkthrough
*  Add mention raid protection feature to auto moderation rules ([link](https://github.com/discordjs/discord.js/pull/9073/files?diff=unified&w=0#diff-62a17cd9a393115953bc426580fd41f674aa1cbac571a2969be2a14f45826311R61), [link](https://github.com/discordjs/discord.js/pull/9073/files?diff=unified&w=0#diff-62a17cd9a393115953bc426580fd41f674aa1cbac571a2969be2a14f45826311R129), [link](https://github.com/discordjs/discord.js/pull/9073/files?diff=unified&w=0#diff-62a17cd9a393115953bc426580fd41f674aa1cbac571a2969be2a14f45826311R191), [link](https://github.com/discordjs/discord.js/pull/9073/files?diff=unified&w=0#diff-0ed1239ac7fe7d69306007aef01362e867629141bd56a925b6dbc1d56e41357cR76), [link](https://github.com/discordjs/discord.js/pull/9073/files?diff=unified&w=0#diff-0ed1239ac7fe7d69306007aef01362e867629141bd56a925b6dbc1d56e41357cR89), [link](https://github.com/discordjs/discord.js/pull/9073/files?diff=unified&w=0#diff-0ed1239ac7fe7d69306007aef01362e867629141bd56a925b6dbc1d56e41357cR240-R250))
* Add safety alerts channel feature to guilds ([link](https://github.com/discordjs/discord.js/pull/9073/files?diff=unified&w=0#diff-9c03fb4d8c1b03db876641d3900a9c08f5a3a30be8d8a3b9840018507243c284R438-R447), [link](https://github.com/discordjs/discord.js/pull/9073/files?diff=unified&w=0#diff-9c03fb4d8c1b03db876641d3900a9c08f5a3a30be8d8a3b9840018507243c284R582-R590), [link](https://github.com/discordjs/discord.js/pull/9073/files?diff=unified&w=0#diff-9c03fb4d8c1b03db876641d3900a9c08f5a3a30be8d8a3b9840018507243c284R857), [link](https://github.com/discordjs/discord.js/pull/9073/files?diff=unified&w=0#diff-9c03fb4d8c1b03db876641d3900a9c08f5a3a30be8d8a3b9840018507243c284R941-R943), [link](https://github.com/discordjs/discord.js/pull/9073/files?diff=unified&w=0#diff-9c03fb4d8c1b03db876641d3900a9c08f5a3a30be8d8a3b9840018507243c284R1247-R1261))

**Status and versioning classification:**

### CODE REVIEW

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by TetieWasTaken — CHANGES_REQUESTED** [review]

**Review by TetieWasTaken — APPROVED** [review]

File: src/structures/Guild.js [hunk 1]
```diff
@@ -1224,6 +1247,21 @@ class Guild extends AnonymousGuild {
...
+   * @param {string} [reason] Reason for changing the guild's safety alerts channel
+   * @returns {Promise<Guild>}
+   * @example
+   * // Edit the guild safety alerts channel
+   * guild.setSafetyAlertsChannel(channel)
+   *  .then(updated => console.log(`Updated guild safety alerts channel to ${guild.safetyAlertsChannel.name}`))
+   *  .catch(console.error);
+   */
+  setSafetyAlertsChannel(safetyAlertsChannel, reason) {
+    return this.edit({ safetyAlertsChannel }, reason);
```
[the remark this answers is not in the document]
Comment (TetieWasTaken) [reply src/structures/Guild.js]: Nevermind, I thought I was looking at the v13 branch. Thank you for reverting it back

File: src/structures/Guild.js [hunk 2]
```diff
@@ -1224,6 +1247,21 @@ class Guild extends AnonymousGuild {
...
 
+  /**
+   * Edits the safety alerts channel of the guild.
+   * @param {?TextChannelResolvable} safetyAlertsChannel The new safety alerts channel
+   * @param {string} [reason] Reason for changing the guild's safety alerts channel
+   * @returns {Promise<Guild>}
+   * @example
+   * // Edit the guild safety alerts channel
+   * guild.setSafetyAlertsChannel(channel)
+   *  .then(updated => console.log(`Updated guild safety alerts channel to ${guild.safetyAlertsChannel.name}`))
```
Comment (Jiralite) [inline src/structures/Guild.js:1258]:
```suggestion
   *  .then(updated => console.log(`Updated guild safety alerts channel to ${updated.safetyAlertsChannel.name}`))
```

File: typings/index.d.ts [hunk 3]
```diff
@@ -5471,6 +5480,8 @@ export type GuildFeatures =
   | 'SEVEN_DAY_THREAD_ARCHIVE'
   | 'PRIVATE_THREADS'
   | 'ROLE_ICONS'
+  | 'AUTO_MODERATION'
+  | 'RAID_ALERTS_ENABLED'
```
Comment (Jiralite) [inline typings/index.d.ts:5484]:
```suggestion
  | 'RAID_ALERTS_ENABLED'
```

File: typings/index.d.ts [hunk 4]
```diff
@@ -5471,6 +5480,8 @@ export type GuildFeatures =
   | 'SEVEN_DAY_THREAD_ARCHIVE'
   | 'PRIVATE_THREADS'
   | 'ROLE_ICONS'
+  | 'AUTO_MODERATION'
```
Comment (Jiralite) [inline typings/index.d.ts:5483]: This already exists.

```suggestion
```

**Review by Jiralite — CHANGES_REQUESTED** [review]

**Review by Jiralite — CHANGES_REQUESTED** [review]
`Guild#features`'s documentation needs updating.

**Review by Jiralite — APPROVED** [review]

**Pull request merged**